static/js/activity.js

```diff
@@ -239,8 +240,14 @@ function actually_update_users() {
     }
 
     var user_info = _.map(users, info_for);
-
-    $('#user_presences').html(templates.render('user_presence_rows', {users: user_info}));
+    if (user_list !== undefined) {
+        // Render right panel partially
+        $.each(user_info, function (index, user) {
+            $('#user_presences').find('[data-email=' + user.email + ']').replaceWith(templates.render('user_presence_row', user));
+        });
+    } else {
+        $('#user_presences').html(templates.render('user_presence_rows', {users: user_info}));
+    }
```

Comment:
Hmm, I think this will not correctly handle resorting the user when their presence info changes.

If you read the code of `sort_users`, you'll see that users are sorted first by presence status and then by name (so present users are always above idle or away users).  So I think this needs to delete the user's row and insert it in the appropriate place in the presence list, rather than just updating the existing entry.